besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java

```diff
@@ -2405,6 +2412,10 @@ private EthNetworkConfig updateNetworkConfig(final NetworkName network) {
         builder.setBootNodes(new ArrayList<>());
       }
       builder.setDnsDiscoveryUrl(null);
+    } else if (dnsDiscoveryUrl != null) {
```

Comment:
I am of the opinion that the CLI flag should override any and all other configurations (default and genesis file).  So I don't think this should be an else if, and it should be after any other attempts to set the value (which would also be after the --network flag is considered).  This _may_ be the correct place but I don't have my IDE handy at the moment.

This should also be validated with a test.  One approach would be to do a config with `--network=goerli` and `--discovery-dns-url=some-bogus-value` and see if the builder is configured with the bogus value.

Reply:
Here it only applies if genesis is null, otherwise if genesis is not null it matches the first condition and skips this else if.
That means that you can override discovery dns url only for known networks as you have to define either a genesis or a network.
For what I understood, known networks all already have their config available (see `DiscoveryConfiguration.java`) and even if you may want to override it, it's not the main expected use case.
The expectation, to me, is that you can use DNS discovery even with a custom network, that means in the case when you use a custom genesis file.
So the option should be taken in account in any cases, not only for predefined networks.